src/features/courses/course-controller.ts

```diff
@@ -1902,41 +1996,44 @@ class CourseController {
         gradeResult,
         submitted,
         timeOfSubmission,
-        problemPath
+        problemPath,
+        saveGrade = true,
     }: SetGradeFromSubmissionOptions): Promise<StudentWorkbook | undefined> => {
         return useDatabaseTransaction(async (): Promise<StudentWorkbook | undefined> => {
+            const studentGradeData = _.cloneDeep(studentGrade.get()) as StudentGradeInterface;
```

Comment:
Should this be a `get plain` instead of deep cloning? It'll still have all the sequelize methods with the deep clone, won't it?

Reply:
I used `plain: true` because I had seen that in answers before, however `plain: true` is actually default
https://sequelize.org/master/class/lib/model.js~Model.html
> When true, the first returned value of aggregateFunction is cast to dataType and returned. If additional attributes are specified, along with group clauses, set plain to false to return all values of all returned rows. Defaults to true

As mentioned in an above comment sequelize model results are actually interfaces with observers for each of the properties where the values are stored in dataValues, however typescript doesn't expose dataValues so you have to call `.get()`

I thought `.get()` was returning an object with sequelize bindings stripped but it's actually just returning the subobject dataValues, so if you change a value in .get it changes in the parent interface however since we didn't go through the interface `.changed()` does not update

That's why I'm deep cloning it, so the actual model is remaining unchanged